AlphaWallet/Browser/Coordinators/ScanQRCodeCoordinator.swift

```diff
@@ -8,36 +8,52 @@ protocol ScanQRCodeCoordinatorDelegate: class {
     func didScan(result: String, in coordinator: ScanQRCodeCoordinator)
 }
 
-final class  ScanQRCodeCoordinator: NSObject, Coordinator {
+final class ScanQRCodeCoordinator: NSObject, Coordinator {
     var coordinators: [Coordinator] = []
     weak var delegate: ScanQRCodeCoordinatorDelegate?
 
-    let navigationController: NavigationController
     lazy var qrcodeController: QRCodeReaderViewController = {
-        let controller = QRCodeReaderViewController(cancelButtonTitle: R.string.localizable.cancel(), chooseFromPhotoLibraryButtonTitle: R.string.localizable.photos())
+        let reader = QRCodeReader(metadataObjectTypes: [AVMetadataObject.ObjectType.qr])
+
+        let controller = QRCodeReaderViewController(
+            cancelButtonTitle: nil,
+            codeReader: reader,
+            startScanningAtLoad: true,
+            showSwitchCameraButton: false,
+            showTorchButton: true,
+            chooseFromPhotoLibraryButtonTitle: R.string.localizable.photos(),
+            bordersColor: Colors.qrCodeRectBorders,
+            messageText: R.string.localizable.qrCodeTitle(),
+            torchTitle: R.string.localizable.light(),
+            torchImage: R.image.light(),
+            chooseFromPhotoLibraryButtonImage: R.image.browse()
+        )
         controller.delegate = self
-        controller.navigationItem.leftBarButtonItem = UIBarButtonItem(barButtonSystemItem: .cancel, target: self, action: #selector(dismiss))
+        controller.title = R.string.localizable.browserScanQRCodeTitle()
+        controller.navigationItem.leftBarButtonItem = UIBarButtonItem(image: R.image.backWhite(), style: .plain, target: self, action: #selector(dismiss))
         controller.delegate = self
         return controller
     }()
-
-    init(
-        navigationController: NavigationController = NavigationController()
-        ) {
+    
+    let navigationController: NavigationController
```

Comment:
- [ ] Can we make it typed `UINavigationController` instead?

- [ ] Can you make this private (if you need to keep it as a property, I think it can just be a local variable)? It can be created and used internally within `ScanQRCodeCoordinator` and not created and passed it from the call site

- [ ] Create another UINavigationController property, this is the one that is passed in and let `ScanQRCodeCoordinator` handle the presenting and dismissing of the private UINavigationController which is holding `QRCodeReaderViewController`.

eg. in `DappBrowserCoordinator`, change it from:

```
let coordinator = ScanQRCodeCoordinator(navigationController: NavigationController())
coordinator.delegate = self
addCoordinator(coordinator)
coordinator.start()
navigationController.present(coordinator.navigationController, animated: true, completion: nil)
```

to:

```
let coordinator = ScanQRCodeCoordinator(navigationController: navigationController) //Pass in `navigationController`
coordinator.delegate = self
addCoordinator(coordinator)
coordinator.start() //move the next line into `start()`
//navigationController.present(coordinator.navigationController, animated: true, completion: nil)
```

I know this isn't consistent across the app, but refer to `SendCoordinator`'s start() and its delegate for an example:

```
navigationController.pushViewController(sendViewController, animated: true)
```

Generally, we should move the codebase towards doing this. The child coordinator is the one responding for showing and hiding the view controller it manages